src/api/routes/tx.ts

```diff
@@ -3,28 +3,41 @@ import { addAsync, RouterWithAsync } from '@awaitjs/express';
 import * as Bluebird from 'bluebird';
 import { DataStore, DbTx } from '../../datastore/common';
 import { getTxFromDataStore } from '../controllers/db-controller';
-import { timeout, waiter, has0xPrefix } from '../../helpers';
+import { waiter, has0xPrefix } from '../../helpers';
 import { validate } from '../validate';
-import * as txSchema from '../../../.tmp/entities/transactions/transaction.schema.json';
-// import * as txSchema from '../entities/transactions/transaction.schema.json';
-import * as txResultsSchema from '../../../.tmp/api/transaction/get-transactions.schema.json';
-// import * as txResultsSchema from '@schemas/api/transaction/get-transactions.schema.json';
+import * as txSchema from '@schemas/entities/transactions/transaction.schema.json';
+import * as txResultsSchema from '@schemas/api/transaction/get-transactions.schema.json';
+import qs = require('qs');
+
+const parsePagingInput = (val: string) => {
+  if (!/^\d+$/.test(val)) {
+    throw new Error('Limit and offset must be numbers');
+  }
+  const parsedInput = parseInt(val, 10);
+  if (isNaN(parsedInput)) throw new Error('Pagination value parsed as NaN');
+  return parsedInput;
+};
 
 export function createTxRouter(db: DataStore): RouterWithAsync {
   const router = addAsync(express.Router());
 
   router.getAsync('/', async (req, res) => {
-    const txs = await db.getTxList();
+    const limit = parsePagingInput(req.query.limit.toString());
+    const offset = parsePagingInput(req.query.offset.toString());
+
+    const { results: txResults, total } = await db.getTxList({ offset, limit });
```

Comment:
I think we'll need an upper bound here to prevent massive requests from slowing down the db layer. Maybe like 200 is good for now? It will likely have to be decreased in production.

Also, if we make this endpoint only responsible for returning an array of tx_id strings, then we can have a much higher upper bound without putting much strain on the system. 

Thoughts?

Reply:
@aulneau if you describe how these list endpoints will be used, specific optimized queries can probably be crafted to serve that purpose

Reply:
This will be useful to get the latest transactions and display them on the explorer. 

It'd be great to be able to fetch `n` of tx_type of all transactions, eg `fetchContractCreationTxs(count)`.

> Also, if we make this endpoint only responsible for returning an array of tx_id strings, then we can have a much higher upper bound without putting much strain on the system.

Both would be ok, ideally we'd richer data, eg type, amount, addresses etc. I don't think there is a way to effectively fetch many tx's from the api, so it would be slow to do on the frontend.

Reply:
@aulneau could you create an issue for an API endpoint like that? Definitely sounds doable.

@kyranjamie Can you add a check here for limit < 200, which returns an error when exceeded?